primitives/election-providers/src/onchain.rs

```diff
@@ -62,9 +69,10 @@ impl<T: Config> ElectionProvider<T::AccountId, T::BlockNumber> for OnChainSequen
 	type DataProvider = T::DataProvider;
 
 	fn elect() -> Result<Supports<T::AccountId>, Self::Error> {
```

Comment:
should elect return the weight used ? on chain it takes full block and multi phase it takes only few weight.

Reply:
Yeah good point, should do

Reply:
Okay -- Doing that would be really more sensible to do when we move these traits to `/frame`. Then, we can actually benchmark the on-chain election and have its real weight being returned. In the current scheme, even knowing the maximum block weight would be kinda funky: the onchain::Config would need to accept a max-block-weight and simply always return it 🤷 

I say we keep this one small and move this stuff to frame asap.

Reply:
#8246